compiler+runtime/src/cpp/jank/read/lex.cpp

```diff
@@ -700,9 +757,20 @@ namespace jank::read
             }
             require_space = false;
             auto const oc(peek());
-            ++pos;
-
-            switch(oc.unwrap_or(' '))
+            char32_t c{};
+            size_t size{};
+            if(oc.is_err())
+            {
+              c = ' ';
+              ++pos;
+            }
```

Comment:
This pattern of replacing an `unwrap_or` with a lot more code shows up a few times. Why not use `unwrap_or(codepoint{ ' ', 1 })` and then remove all of the extra code? We should be working with `codepoint` structs here, not individual `char32_t c` and `size_t size` locals over and over.